main.js

```diff
@@ -0,0 +1,329 @@
+    let forecastForDay = fiveDayForecastsDOM[i];
+    let template = `
+      <div class="forecast-infobox column">
+        <h5 class="weather">${forecastForDay.clouds}</h5>
+        <h5 class="temperature">${forecastForDay.temp}</h5>
+        <div id="small-weather-icon">${selecSmallWeatherIcon(forecastForDay.clouds)}</div>
+        <h5 class="day-of-week">${forecastForDay.dayOfWeek}</h5>
+      </div>`;
+    
+    let htmlObject = document.createElement('div');
+    htmlObject.className = 'forecast';
+    htmlObject.innerHTML = template;
+  
+    document.querySelector('.five-day-forecast').insertAdjacentElement('beforeend', htmlObject);
+  }
+};
+
+/**
+ * 
+ * @param {JSON} data 
+ * @returns {Array} of day objects for five day forecast received from JSON API call
+ */
+let getFiveDayForecastDayObjArr = function(data){
+  let fiveDayForecastDays = [];
+  let allObjForecastEntriesArr = [];
+
+  data.list.forEach(function(item){
+    let dayOfWeek = getNameOfDay(item);
+    let dayObj = {
+      city: data.city.name,
+      dayOfWeek: dayOfWeek,
+      temp: Math.floor(item.main.temp),
+      clouds: item.weather[0].main
+    }
+    allObjForecastEntriesArr.push(dayObj);
+  });
+
+  fiveDayForecastDays = function(objArray){
+    return objArray.reduce(function(acc, dayOfWeek){
+      let exists = acc.find(function (item){
+        return dayOfWeek.dayOfWeek === item.dayOfWeek;
+      });
+      if(!exists){
+        acc.push(dayOfWeek);
+      }
+      return acc;
+    }, []);
+  }
+  return fiveDayForecastDays(allObjForecastEntriesArr);
+}
+
+/**
+ * 
+ * @param {JSON} item - item object from JSON object promise
+ * @returns {String} - string name of day of week from JSON object
+ */
+function getNameOfDay (item){
+  let dayNumIndex = new Date(item.dt*1000).getDay();
+  return dayNameNumberIndicesObj[dayNumIndex];
+}
+
+/**
+ * 
+ * @param {Array} tempsArray 
+ * @returns {Number} - average of all number items in @param tempsArray
+ */
+
+function avgTempForDay(tempsArray){
+  let avgTemp = tempsArray.reduce(function(sum, temp){
+    return sum + temp;
+  }, 0);
+  return Math.floor(avgTemp/tempsArray.length);
+}
+
+/**
+ * 
+ * @param {Array} cloudsArray 
+ * @returns {Number} - number for highest count item in @param {Array} cloudsArray 
+ */
+function avgCloudCoverForDay(cloudsArray){
+  let forecastObjCount = cloudsArray.reduce(function(acc, forecast){
+    if(acc.hasOwnProperty(forecast)){
+      acc[forecast] += 1;
+    } else {
+      acc[forecast] = 1;
+    }
+    return acc;
+  }, {});
+
+  let highestCount = Math.max(...Object.values(forecastObjCount));
+
+  return Object.keys(forecastObjCount).find(function(key){
+    return forecastObjCount[key] === highestCount;
+  });
+}
+
+/**
+ * 
+ * @param {String} forecastSky 
+ * @returns {String} - string representation of DOM element for weather forecast icon image representation - Big version
+ */
+let selectBigWeatherIcon = function(forecastSky){
+  switch(true) {

```

Comment:
You should do switch(forecastSky) on this - please rereview the mdn for switch: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch